internal/declcfg/load.go

```diff
@@ -135,20 +137,24 @@ func readYAMLOrJSON(r io.Reader) (*DeclarativeConfig, error) {
 		case schemaPackage:
 			var p Package
 			if err := json.Unmarshal(doc, &p); err != nil {
-				return nil, fmt.Errorf("parse package: %v", err)
+				errs = append(errs, fmt.Errorf("parse %s %s: %v", in.Schema, in.Name, err))
```

Comment:
If schema and name are empty strings (which is definitely possible if we're hitting an unmarshal error), this error message will look like `parse    : error`

Maybe something like this. I don't love the square brackets, so if you have a better idea, I'm all ears.
```suggestion
				errs = append(errs, fmt.Errorf("parse [schema:%q name:%q]: %v", in.Schema, in.Name, err))
```

Reply:
It might be helpful to have the processed filenames on the errors in the Walk function, where schema and name might be empty.

https://github.com/operator-framework/operator-registry/blob/78f27b39dc098c2e56681d9a3d8e786aae95d4fe/internal/declcfg/load.go#L39

Reply:
At the point these errors are being constructed, at least `Meta.Schema` is known, so the error will always at least look like `parse olm.<package|bundle>  : error`, so `parse %s [name:%q]: %v` would be reasonable. Thoughts?

>It might be helpful to have the processed filenames on the errors in the Walk function

@ankitathomas I'll add a path component to some errors in that function.

Reply:
Good point about schema being known. That actually simplifies this IMO. How about just:

```
return fmt.Errorf("parse %s name=%q: %v", m.Schema, name, err)
```

I think subbing in `"unknown"` when name is empty could be more confusing than just leaving it blank.